source/module_base/test/tool_check_test.cpp

```diff
@@ -0,0 +1,98 @@
+#include "../tool_check.h"
+#include "gtest/gtest.h"
+#include <fstream>
+#include <cstdio>
+
+/************************************************
+*  unit test of class ToolCheckTest
+***********************************************/
+
+/**
+ * - Tested Function
+ *   - ModuleBase::CHECK_NAME
+ *	- check the next input from ifs is std::string
+ *
+ *   - ModuleBase::CHECK_INT
+ *	- check the next input from ifs is int
+ *
+ *   - ModuleBase::CHECK_DOUBLE
+ *	- check the next input from ifs is double
+ *
+ *   - ModuleBase::CHECK_STRING
+ *	- check the next input from ifs is string
+ *
+ */
+
+
+class ToolCheckTest : public testing::Test
+{
+    protected:
+        std::ofstream ofs;
+	std::ifstream ifs;
+	// define std::string, int, double variables
+	std::string name="abaqus";
+	int ecut = 100;
+	double occupation = 0.23;
+	std::string caltype="nscf";
+	// quit is the swith to control performance of function
+	bool quit=false;
```

Comment:
Maybe `no_quitting` for variable name is more appropriate? It would become `ModuleBase::CHECK_NAME(ifs,"abacus",no_quitting);`.